### ISSUE TITLE
Mithril IE9 m.route(url...) using #

### ISSUE THREAD

**Issue by LorenzV** [lhorie/mithril.js#320]
In the docs i saw that IE9(& lower) m.route(...) doesn't work because of pushstate support.
My app uses # for urls, and it seems you can just create a quick work-around for getting this to work in IE9 without refreshing. (don't know about other IE's)
Maybe you could add this work-around in mithril itself, or something similar. I don't really need it though, just asking your opinion on this, if it's bad idea or not.

So here it is, i used jquery for adding the html + setting the Attr. I haven't tested this version.
```
document.body.innerHTML += '<a id="mithrilVirtualLink" href="#"></a>';
var mRedirect = function(url){
    document.getElementById('mithrilVirtualLink').setAttribute('href', '#'+url);
    document.getElementById('mithrilVirtualLink').click();
}
```

**Comment by jupl** [comment]
I have to end up doing this as a (hopefully) temp hack for IE9 not rendering the view on `m.route`.

```js
// Called only if IE<=9
function redirect(url) {
  window.location.href = "#" + url;
  setTimeout(m.redraw, 0);
}
```

**Comment by jupl** [comment]
The big problem I'm noticing that it may be in IE9 is that hash changes are not being picked up by Mithril's routing. For instance, [in this fiddle](http://jsfiddle.net/3sp4jasb/5/) I have to set the hash in advance before bootstrapping the routes so it shows something. Without that the page does not render at all in IE9. Chrome/Firefox/Safari/IE11 works as expected. Even with setting the hash in IE9 clicking on the links do change the hash but does not cause page updates.

This is also apparent in https://gist.github.com/lhorie/681e23f3f0014b16aa6d. I tried this and in IE9 the `m.module` part renders fine whereas the `m.route` part does not render. No errors are thrown when debugging.

I tried with 0.1.30, 0.1.22, and 0.1.24 and neither worked for me.
